### ISSUE TITLE
Finish inkey()/inkey_ex() migration

### ISSUE THREAD

**Issue by flambard-took** [mangband/mangband#1265]
The work done by Billsey to support newer V Z-Term (#944) left one feature in a TODO state: the newer `inkey_ex()` functions are mostly discarded in favor of our old, trusty `inkey()`.

This was nice, because it kept everything working, and if we're to migrate completely we'll likely break some things. But we do need to migrate, because we **do** want to actually use the new functions.

Here's what's going on (I'm comparing to V309):

function | In V:| In MAng:
----|--------|--------
char inkey() | thin wrapper around `inkey_ex` | main function to grab input
event_type inkey_aux() | main function to grab input | same, but mostly unused
event_type inkey_ex() | return next input event, as a struct | same, but only returns `char`

In other words, in V, inkey() calls inkey_ex(), which calls inkey_aux(). All the way, `event_type` is returned, only to be discarded at inkey(). If you want single char, call inkey(), if you want extended event, call inkey_ex(). In MAng, inkey_ex() can't return extended information, so there is no way to get it; inkey() doesn't call inkey_ex(), but does it's own processing.

Also, the `macro complete` char used in V is `30`, yet it is `29` in MAng.

Also, the `parse_slash` mode is completely removed from V, so there's no special handling of char `28`.

For this ticket, I'm keeping the char `28` hack, yet am planning to upgrade `macro complete` char from 29 to 30, to match V.
